src/rootcheck/check_rc_dev.c

```diff
@@ -37,9 +37,18 @@ static int read_dev_file(const char *file_name)
 
     else if (S_ISREG(statbuf.st_mode)) {
         char op_msg[OS_SIZE_1024 + 1];
+        const char op_msg_fmt[] = "File '%*s' present on /dev. Possible hidden file.";
+
+        const int size = snprintf(NULL, 0, op_msg_fmt, (int)strlen(file_name), file_name);
+
+        if (size >= 0 && (size_t)size < sizeof(op_msg)) {
+            snprintf(op_msg, sizeof(op_msg), op_msg_fmt, (int)strlen(file_name), file_name);
+        }
+        else {
+            const unsigned int surplus = size - sizeof(op_msg) + 1;
+            snprintf(op_msg, sizeof(op_msg), op_msg_fmt, (int)(strlen(file_name) - surplus), file_name);
+        }
```

Comment:
If size is negative, calculation of surplus could be wrong
```suggestion
        if (size >= 0) { 
            if ((size_t)size < sizeof(op_msg)) {
                snprintf(op_msg, sizeof(op_msg), op_msg_fmt, (int)strlen(file_name), file_name);
            } else {
                const unsigned int surplus = size - sizeof(op_msg) + 1;
                snprintf(op_msg, sizeof(op_msg), op_msg_fmt, (int)(strlen(file_name) - surplus), file_name);
            }
        }
```